### ISSUE TITLE
AssetStore is only 2018.4.3 and some problem

### ISSUE THREAD

**Issue by Personuo** [gamedevware/charon-unity3d#6]
Thank you for this asset. I found this asset just now and send 5 star
But, I found the charon version is 2018.4.3 when I download from AssetStroe
Can you upload new version to AssetStore or github  releases?

And Restore function has bug.

**Comment by deniszykov** [comment]
Hi @cn914287516! Thanks for your report. I have fixed 'restore' button in 663426efaea9a0af75db059f36a19f9a7b6c334a commit and submitted version 2019.1 in asset store. You can update plugin from Unity. Choose 'Tools' -> 'Charon' -> 'Check for updates...'.

> When I use generate source Code ...

I have found and fixed bugs with .asset generation. Fixes in 2019.1

> I want to Read GameData from Resource.Load ,how to do?

You can select "Generate Code and Asset" in code generation options and load generated asset with `Resources.Load<GameData>(path)```. 
Or you can select 'Generate Code' and [load gamedata](https://gamedevware.com/docs/pages/viewpage.action?pageId=1277972) from [Stream](https://docs.microsoft.com/ru-ru/dotnet/api/system.io.stream?view=netframework-4.7.2):
``` 
var fileStream = File.OpenRead("gamedata.json");
var gameData = new GameData(fileStream, GameData.Format.Json);
``` 

> How can I encryption data?

This is beyond scope of this tool. You can download/decompress/encrypt/patch data stream right before passing it to  `new GameData(fileStream, ...)`

**Comment by deniszykov** [comment]
Also, thanks for 5 stars :-)

**Comment by Personuo** [comment]
Thx for relply.
And when I select "Generate Code and Asset",for example class name "MyGameData"
It's will generate one c# script that name is "MyGameData.cs" and Generate MyGameData.Asset.
But MyGameData.Asset lost MyGameData.cs depend;
I must Move MyGameData:ScriptableObject to the top of "MyGameData.cs".

**Comment by deniszykov** [comment]
> I must Move MyGameData:ScriptableObject to the top of "MyGameData.cs".

I have generated asset at `Assets/Resources/GameData.asset` and successfully loaded it with following code `this.GameData = Resources.Load<RpgGameData>("GameData");`. I made this tests with project from [Example](https://github.com/deniszykov/charon-unity3d/tree/master/Examples/RpgGameExample) folder.

[Here](https://monosnap.com/direct/6Yi0LINqWAANSvQq04u7V1iwOfNibL) is screenshot.

Can you provide more details about error you getting?

**Comment by Personuo** [comment]
> > I must Move MyGameData:ScriptableObject to the top of "MyGameData.cs".
> 
> I have generated asset at `Assets/Resources/GameData.asset` and successfully loaded it with following code `this.GameData = Resources.Load<RpgGameData>("GameData");`. I made this tests with project from [Example](https://github.com/deniszykov/charon-unity3d/tree/master/Examples/RpgGameExample) folder.
> 
> [Here](https://monosnap.com/direct/6Yi0LINqWAANSvQq04u7V1iwOfNibL) is screenshot.
> 
> Can you provide more details about error you getting?

my mean is GameData.Asset will lost GameData.cs depend
click GameData.Asset and look Inspector.
[Script is missing](https://monosnap.com/file/rDnnZe7h02PoOqVVZEJgvihJhoKMCO#)

If your want  asset link a ScriptableObject script.
the ScriptableObject must be the first class in script.[pic](https://monosnap.com/file/Bg6V8Odj3ltOZhJ1lFavBUtPoCkCBF#)

**Comment by deniszykov** [comment]
Which Unity version do you use? I have no errors with code dependency in 2018.2.

Anyway, I changed class order in generated code. Just do  'Tools' -> 'Charon' -> 'Check for updates...' and update to 2019.1.1.

**Comment by Personuo** [comment]
My unity version is 2018.3,and I try use unity 2018.2 to test and it's all ok.
So I think this is the problem of unity 2018.3.

About Charon new verion 2019.1, I am downloading, the server is very slow

**Comment by Personuo** [comment]
> Which Unity version do you use? I have no errors with code dependency in 2018.2.
> 
> Anyway, I changed class order in generated code. Just do 'Tools' -> 'Charon' -> 'Check for updates...' and update to 2019.1.1.

I update Charon to 2019.1.1 . [pic](https://monosnap.com/file/72Wzg99WpcOt2YdwOZ9UDiHxAb2gQV)
And I try to generate code and get a [error](https://monosnap.com/file/pDDnbqpLTlChqZptW8hE0z4m9T64tg#) .
it generate an empty script.
 
By the way, before I used 2018.4.3 was successful.

Unity version is 2018.2.20

**Comment by deniszykov** [comment]
Hi! Thank for your report. I have found source of this error - missing ProjectSettings in gamedata. I have made fixes in 2019.1.2. You can workaround this problem by opening file in editor, selecting 'Project Settings' and clicking 'Save'. Or update to 2019.1.2

**Comment by Personuo** [comment]
Ok It's work ,but bad news is" generate C sharp code and asset" is error:Assets/Scripts/GameData.cs(59,0): error CS1525: Unexpected symbol `using'.

**Comment by deniszykov** [comment]
I think to take some time to make proper release without hotfixing 'on a fly'. ETA few days.

-

**Comment by Personuo** [comment]
> I think to take some time to make proper release without hotfixing 'on a fly'. ETA few days.

Very good, please let me know when you are done, thank you

**Comment by Personuo** [comment]
> My unity version is 2018.3,and I try use unity 2018.2 to test and it's all ok.
> So I think this is the problem of unity 2018.3.
> 
> About Charon new verion 2019.1, I am downloading, the server is very slow

Now,This is a unity bug,and 2018.3.10f is fixed.

**Comment by deniszykov** [comment]
> Now,This is a unity bug,and 2018.3.10f is fixed.

Still, I'm making changes in new version to generate additional class for asset.

**Comment by deniszykov** [comment]
Fixed in 2019.2.3
